examples/pipeline/dummy_entity_linking.py

```diff
@@ -0,0 +1,69 @@
+# coding: utf-8
+"""Demonstrate how to build a simple knowledge base and run an Entity Linking algorithm.
+Currently still a bit of a dummy algorithm: taking simply the entity with highest probability for a given alias
+"""
+import spacy
+from spacy.kb import KnowledgeBase
+
+
+def create_kb(vocab):
+    kb = KnowledgeBase(vocab=vocab)
+
+    # adding entities
+    entity_0 = "Q1004791"
+    print("adding entity", entity_0)
+    kb.add_entity(entity_id=entity_0, entity_name="Douglas", prob=0.5)
+
+    entity_1 = "Q42"
+    print("adding entity", entity_1)
+    kb.add_entity(entity_id=entity_1, entity_name="Douglas Adams", prob=0.5)
+
+    entity_2 = "Q5301561"
+    print("adding entity", entity_2)
+    kb.add_entity(entity_id=entity_2, entity_name="Douglas Haig", prob=0.5)
```

Comment:
What's the significance of the `entity_name` vs. the `entity_id`? Is it just the human-readable version? Is it supposed to map to the `Span.text`?

(I'm not an entity linking expert, so there's a chance I'm missing something obvious here.)

Reply:
It's like the one official standard human-readable name for an entity, like entity_id="Q42" and entity_name="Douglas Adams". But it's not unique, while entity_id is. I was thinking about whether we need this specifically, but from experience I think we're going to need this when printing things etc, humans don't usually like reading unique machine IDs...

Reply:
Ah okay, yes, that makes sense. The only potential problem I see here is that a user might falsely assume that the `entity_name` has significance for the result, or confuse it with an alias. Because in this example, the entity name and alias are identical ("Douglas Adams"), but the alias is what actually matters (right?).

Reply:
Yea, that's right. Worse even, currently the `entity_name` isn't even recorded as an alias because aliases are recorded alias-by-alias to make sure the sum of their prior probabilities don't exceed 1. So yes, it's confusing :-(

Reply:
I think we'd probably prefer to force `name` to be unique, and then use it as the ID. For consistency with the rest of the API, we're always going to need to accept both string and hash versions of these things. It gets pretty unwieldy if you have a name and an ID, and then the "integer ID of the name" (how we describe the hash in the docs) isn't the same as the ID.

It's a pity we don't have the `P(mention | candidate)` distribution. If we did we could just use that to get the canonical mention for the entity, i.e. whatever people usually call it, which could be used as the human-readable name.

Reply:
ok, so I remove (non-unique) `entity_name`, and rename `entity_id` to (unique) `entity_name` ? or just `entity` (as it was initially, I think) ?

I was also thinking along those lines @honnibal. Those probabilities would be too detailed perhaps, but you could imagine each alias having a type, such as "abbreviation" or "standard_name". But then you would be inclined to add the aliases entity-by-entity instead of mention-by-mention.

Reply:
Another option would be to sort of extract a human-readible name for each entity, given the final annotations in the document(s). So if I processed all these texts and Q42 is most often linked to the alias "A.D.", then that becomes the "standard" string of Q42 in those docs. It is at least closest to the actual processed text & its user.

Reply:
update: simplified things: https://github.com/explosion/spaCy/pull/3459/commits/8814b9010d139f92bc817378eace25e24a817b7e

Reply:
> ok, so I remove (non-unique) entity_name, and rename entity_id to (unique) entity_name ? or just entity (as it was initially, I think) ?

Yes, I think so.

> Another option would be to sort of extract a human-readible name for each entity, given the final annotations in the document(s).

Well, maybe. I actually think the unique ID won't be too bad; I mean for Wikipedia pages it's fine, and for other knowledge bases, people basically are responsible for coming up with a scheme that's useful to them. If something's lacking we could develop it later.